models/v3/msgs/msgs_test.go

```diff
@@ -519,6 +521,376 @@ func TestNewEventMetaData(t *testing.T) {
+			parentHomeTenantID:     "parent-tenant-1",
+			parentResourceTenantID: "parent-resource-tenant-1",
+			resourceConfigs: []struct{ homeTenantID, resourceHomeTenantID string }{
+				{homeTenantID: "", resourceHomeTenantID: ""},
+				{homeTenantID: "", resourceHomeTenantID: ""},
+			},
+			expectError:   true,
+			errorContains: "Event.Data: .Resources[0].HomeTenantID \"\" must match Data.HomeTenantID \"parent-tenant-1\"",
+		},
+		{
+			name:                   "Parent empty, all children set with same values - should error (strict validation)",
+			parentHomeTenantID:     "",
+			parentResourceTenantID: "",
+			resourceConfigs: []struct{ homeTenantID, resourceHomeTenantID string }{
+				{homeTenantID: "child-tenant-1", resourceHomeTenantID: "child-resource-tenant-1"},
+				{homeTenantID: "child-tenant-1", resourceHomeTenantID: "child-resource-tenant-1"},
+			},
+			expectError:   true,
+			errorContains: "Event.Data: .Resources[0].HomeTenantID \"child-tenant-1\" must match Data.HomeTenantI",
+		},
+		{
+			name:                   "Parent empty, children have mixed values - should error",
+			parentHomeTenantID:     "",
+			parentResourceTenantID: "",
+			resourceConfigs: []struct{ homeTenantID, resourceHomeTenantID string }{
+				{homeTenantID: "child-tenant-1", resourceHomeTenantID: "child-resource-tenant-1"},
+				{homeTenantID: "", resourceHomeTenantID: ""},
+				{homeTenantID: "child-tenant-1", resourceHomeTenantID: "child-resource-tenant-1"},
+			},
+			expectError:   true,
+			errorContains: "HomeTenantID",
+		},
+		{
+			name:                   "Some children have tenant IDs, others don't - should error",
+			parentHomeTenantID:     "",
+			parentResourceTenantID: "",
+			resourceConfigs: []struct{ homeTenantID, resourceHomeTenantID string }{
+				{homeTenantID: "tenant-1", resourceHomeTenantID: "resource-tenant-1"},
+				{homeTenantID: "", resourceHomeTenantID: "resource-tenant-1"},
+			},
+			expectError:   true,
+			errorContains: "HomeTenantID",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var resources []types.NotificationResource
+			for i, config := range test.resourceConfigs {
+				resource := createTestResource(i+1, config.homeTenantID, config.resourceHomeTenantID)
+				resources = append(resources, resource)
+			}
+
+			notifications := Notifications{
+				ResourceLocation:     "eastus",
+				PublisherInfo:        "Microsoft.Test",
+				HomeTenantID:         test.parentHomeTenantID,
+				ResourceHomeTenantID: test.parentResourceTenantID,
+				Data:                 resources,
+			}
+
+			_, event, err := notifications.toEvent()
+			if err != nil && !test.expectError {
+				t.Errorf("toEvent() failed unexpectedly: %v", err)
+				return
+			}
+
+			err = event.Validate()
+			if test.expectError {
+				if err == nil {
+					t.Errorf("Expected error but got none")
+					return
+				}
+				if !strings.Contains(err.Error(), test.errorContains) {
```

Comment:
No, we don't check for error strings.  Its either binary (we check if there is or isn't an error) or if we REALLY need to, we check by error type.

Never string matches.

Reply:
And anywhere else you might be doign this.

Reply:
Adjusted to only check for error, using a switch statement now too for parity with other tests in the file.